utility/unit_of_work/uow_leader_test.go

```diff
@@ -0,0 +1,103 @@
+package unit_of_work
+
+import (
+	"fmt"
+	"math/big"
+	"reflect"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/pokt-network/pocket/shared/modules"
+	mockModules "github.com/pokt-network/pocket/shared/modules/mocks"
+	"github.com/pokt-network/pocket/shared/utils"
+)
+
+var DefaultStakeBig = big.NewInt(1000000000000000)
+
+func Test_leaderUtilityUnitOfWork_CreateProposalBlock(t *testing.T) {
+	t.Helper()
+
+	type fields struct {
+		leaderUOW func(t *testing.T) *leaderUtilityUnitOfWork
+	}
+	type args struct {
+		proposer   []byte
+		maxTxBytes uint64
+	}
+	tests := []struct {
+		name          string
+		args          args
+		fields        fields
+		wantStateHash string
+		wantTxs       [][]byte
+		wantErr       bool
+	}{
+		{
+			name: "should revert a failed block proposal",
+			args: args{},
+			fields: fields{
+				leaderUOW: func(t *testing.T) *leaderUtilityUnitOfWork {
+					ctrl := gomock.NewController(t)
+
+					mockUtilityMod := newDefaultMockUtilityModule(t, ctrl)
+					mockrwcontext := newDefaultMockRWContext(t, ctrl)
+
+					mockrwcontext.EXPECT().ComputeStateHash().Return("", fmt.Errorf("rollback error"))
```

Comment:
NIT: I don't fully understand the choice whitespacing & grouping of functions here, but not a big deal.

I would've personally gone with:
```go
mockbus := mockModules.NewMockBus(ctrl)

mockUtilityMod := newDefaultMockUtilityModule(t, ctrl)
mockbus.EXPECT().GetUtilityModule().Return(mockUtilityMod).AnyTimes()
	
mockrwcontext := newDefaultMockRWContext(t, ctrl)
mockrwcontext.EXPECT().ComputeStateHash().Return("", fmt.Errorf("rollback error"))

luow := NewLeaderUOW(0, mockrwcontext, mockrwcontext)
luow.SetBus(mockbus)
```